tests/conftest.py

```diff
@@ -2144,3 +2144,31 @@ def arrow_table_with_only_nulls(pa_schema: "pa.Schema") -> "pa.Table":
     import pyarrow as pa
 
     return pa.Table.from_pylist([{}, {}], schema=pa_schema)
+
+
+@pytest.fixture()
+def catalog_rest() -> Catalog:
+    return load_catalog(
+        "local",
+        **{
+            "type": "rest",
+            "uri": "http://localhost:8181",
+            "s3.endpoint": "http://localhost:9000",
+            "s3.access-key-id": "admin",
+            "s3.secret-access-key": "password",
+        },
+    )
+
+
+@pytest.fixture()
+def catalog_hive() -> Catalog:
```

Comment:
@frankliee Thanks for working on this! We also have `session_catalog` and `session_hive_catalog` in `conftest` 
https://github.com/apache/iceberg-python/blob/aa4654368e54bf84933279179519b299b5910493/tests/conftest.py#L1997-L2022
and I think they can be unified with these too (using scope `session`). WDYT?

Reply:
> @frankliee Thanks for working on this! We also have `session_catalog` and `session_hive_catalog` in `conftest`
> 
> https://github.com/apache/iceberg-python/blob/aa4654368e54bf84933279179519b299b5910493/tests/conftest.py#L1997-L2022
> 
> 
> and I think they can be unified with these too (using scope `session`). WDYT?

Nice suggestion, I will fix them in this PR after a while.